spark/src/main/scala/org/apache/spark/sql/delta/RowId.scala

```diff
@@ -110,7 +110,11 @@ object RowId {
    * Extracts the high watermark of row IDs from a snapshot.
    */
   private[delta] def extractHighWatermark(snapshot: Snapshot): Option[Long] =
-    RowTrackingMetadataDomain.fromSnapshot(snapshot).map(_.rowIdHighWaterMark)
+    if (isEnabled(snapshot.protocol, snapshot.metadata)) {
```

Comment:
This should be `isSupported(snapshot.protocol)`. `baseRowIds` must already be set when `enableRowTracking` is `false` (but the feature is enabled). Returning `None` here will result in overlapping row ids being assigned.

Reply:
Thanks, this was a mistake.  Fixed now :)

Reply:
No problem :D. Seems like we are missing some test coverage here. I'll prioritize adding a test.